src/plugins/ssh-agent/index.ts

```diff
@@ -0,0 +1,146 @@
+/** Copyright © 2024-present P0 Security
+
+This file is part of @p0security/cli
+
+@p0security/cli is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 3 of the License.
+
+@p0security/cli is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with @p0security/cli. If not, see <https://www.gnu.org/licenses/>.
+**/
+import { print2 } from "../../drivers/stdio";
+import { AgentArgs, SshAgentEnv } from "./types";
+import { SpawnOptionsWithoutStdio, spawn } from "node:child_process";
+
+const AUTH_SOCK_MESSAGE = /SSH_AUTH_SOCK=(.+?);/;
+const AGENT_PID_MESSAGE = /SSH_AGENT_PID=(\d+?);/;
+
+/** Spawns a subprocess with given command, args, and options.
+ * May write content to its standard input.
+ * Stdout and stderr of the subprocess is printed to stderr in debug mode.
+ * The returned promise resolves or rejects with the exit code. */
+const asyncSpawn = async (
+  { debug }: AgentArgs,
+  command: string,
+  args?: ReadonlyArray<string>,
+  options?: SpawnOptionsWithoutStdio,
+  writeStdin?: string
+) =>
+  new Promise<number>((resolve, reject) => {
+    const child = spawn(command, args, options);
+
+    if (writeStdin) {
+      if (!child.stdin) return reject("Child process has no stdin");
+      child.stdin.write(writeStdin);
+    }
+
+    child.stdout.on("data", (data) => {
```

Comment:
DRY: could we share the callback function somehow? 🤔

Reply:
I tried but I think it would require "mutating" a string inside a function and the mutation should be reflected outside. I would probably need an object or class to do so but then it becomes too much code imo.

Reply:
understandable 👍